spec/controllers/submissions_controller_spec.rb

```diff
@@ -400,52 +400,88 @@
                             release_results: 'true' }
           is_expected.to respond_with(:success)
         end
-
-        it 'should send an email to one grouping if only one is selected' do
-          allow(Assignment).to receive(:find) { @assignment }
-          expect do
-            post_as @admin,
-                    :update_submissions,
-                    params: { assignment_id: 1,
-                             groupings: ([] << @assignment.groupings).flatten,
-                             release_results: 'true' }
-          end.to change { ActionMailer::Base.deliveries.count }.by(1)
-        end
-        it 'should send emails for every grouping selected if more than one are selected' do
-          allow(Assignment).to receive(:find) { @assignment }
-          @other_grouping = create(:grouping, assignment: @assignment)
-          @other_membership = create(:student_membership, membership_status: 'inviter', grouping: @other_grouping)
-          @other_grouping.group.access_repo do |repo|
-            txn = repo.get_transaction('test')
-            path = File.join(@assignment.repository_folder, 'file1_name')
-            txn.add(path, 'file1 content', '')
-            repo.commit(txn)
-            # Generate submission
-            submission = Submission.generate_new_submission(@other_grouping, repo.get_latest_revision)
-            result = submission.get_latest_result
-            result.marking_state = Result::MARKING_STATES[:complete]
-            result.save
-            submission.save
+        context 'with one grouping selected' do
+          it 'sends an email to the student if only one student exists in the grouping' do
+            expect do
+              post_as @admin,
+                      :update_submissions,
+                      params: { assignment_id: @assignment.id,
+                                groupings: ([] << @assignment.groupings).flatten,
+                                release_results: 'true' }
+            end.to change { ActionMailer::Base.deliveries.count }.by(1)
+          end
+          it 'sends an email to every student in a grouping if it has multiple students' do
+            @another_membership = create(:student_membership, membership_status: 'inviter', grouping: @grouping)
```

Comment:
Since you're making updates to these tests and adding new ones with the same form: this variable doesn't need to be an instance variable, you can make it local. (And check for this throughout.)